### PR TITLE
Add ETH-ETH atomic swap integration test using same contract on same blockchain

### LINKED ISSUE #818: Create transactions for simple ETH-ETH atomic swap (one chain)
Builds on #817 

Add some custom ethereum transactions and execute an atomic swap.

Use two different copies of the same contract on one ganche chain, so they think they are separate.

As this would be second specific contract call (after erc20 example), this should include refactoring the generic "abi contract call" transaction to work for the more general case.
 
Note: we could also start up to local ganache chains and swap between them if not too much overhead to set up

### PR DESCRIPTION
**Pull request by willclarktech**
Closes #818

### CODE REVIEW

File: packages/iov-ethereum/src/integrationtests/atomicswap.spec.ts [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+    const wallet = profile.addWallet(Secp256k1HdWallet.fromMnemonic(mnemonic));
+    const connection: AtomicSwapConnection = await EthereumConnection.establish(
+      testConfig.base,
+      testConfig.connectionOptions,
+    );
+    const coinType = 60;
+    const account = 0;
+    const change = 0;
+    const path1 = HdPaths.bip44(coinType, account, change, addressIndex1);
+    const path2 = HdPaths.bip44(coinType, account, change, addressIndex2);
```
Comment (webmaster128) [inline packages/iov-ethereum/src/integrationtests/atomicswap.spec.ts]: `const path1 = HdPaths.ethereum(addressIndex1);`

File: packages/iov-ethereum/src/integrationtests/atomicswap.spec.ts [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+
+  public async claimFromRevealedPreimage(): Promise<void> {
+    // find claim
+    const claimSwaps = await this.connection.getSwaps({ sender: this.sendAddress });
+    const claim1 = claimSwaps[claimSwaps.length - 1];
+    expect(claim1.kind).toEqual(SwapProcessState.Claimed);
+
+    const offerSwaps = await this.connection.getSwaps({ recipient: this.receiveAddress });
+    const offer = offerSwaps[offerSwaps.length - 1];
+    expect(offer.kind).toEqual(SwapProcessState.Open);
```
Comment (webmaster128) [inline packages/iov-ethereum/src/integrationtests/atomicswap.spec.ts]: I'm not super happy about the expect() calls inside of the `Actor` class. I know this works and why and – shame on me – I also did this in the other file. But I think this belongs into the `it` body directly or should be converted into an exception.

File: packages/iov-ethereum/types/integrationtests/atomicswap.spec.d.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export {};
```
Comment (webmaster128) [inline packages/iov-ethereum/types/integrationtests/atomicswap.spec.d.ts]: When you change the move-types script in `packages/iov-ethereum/package.json` to (copied from @iov/core)
```
"move-types": "shx rm -r ./types/* && shx mv build/types/* ./types && shx rm ./types/*.spec.d.ts && shx rm ./types/**/*.spec.d.ts",
```

this empty file should disappear. Unfortunatly ** does not work  for no subfolder, so there are two versions of that script.

File: packages/iov-ethereum/src/integrationtests/atomicswap.spec.ts (hunk 2 above)
Comment (willclarktech) [reply packages/iov-ethereum/src/integrationtests/atomicswap.spec.ts]: I definitely agree with this, I was just following the pattern of `iov-core`.

**Pull request merged**